core/domain/feedback_services_test.py

```diff
@@ -209,41 +228,90 @@ def setUp(self):
+            0)
+        self.process_and_flush_pending_tasks()
+
     def test_get_all_threads(self):
         # Create an anonymous feedback thread
-        expected_thread_dict = {
-            'status': u'open',
-            'state_name': u'a_state_name',
-            'summary': None,
-            'original_author_username': None,
-            'subject': u'a subject'
-        }
         feedback_services.create_thread(
-            self.EXP_ID, expected_thread_dict['state_name'], None,
-            expected_thread_dict['subject'], 'not used here')
+            self.EXP_ID_1, self.EXPECTED_THREAD_DICT['state_name'], None,
+            self.EXPECTED_THREAD_DICT['subject'], 'not used here')
 
-        threads = feedback_services.get_all_threads(self.EXP_ID, False)
+        threads = feedback_services.get_all_threads(self.EXP_ID_1, False)
         self.assertEqual(1, len(threads))
-        self.assertDictContainsSubset(expected_thread_dict,
+        self.assertDictContainsSubset(self.EXPECTED_THREAD_DICT,
                                       threads[0].to_dict())
 
+        self.EXPECTED_THREAD_DICT_VIEWER['original_author_username'] = (
+            self.VIEWER_USERNAME)
+
         # Viewer creates feedback thread
-        expected_thread_dict = {
-            'status': u'open',
-            'state_name': u'a_state_name_second',
-            'summary': None,
-            'original_author_username': self.VIEWER_USERNAME,
-            'subject': u'a subject second'
-        }
         feedback_services.create_thread(
-            self.EXP_ID, expected_thread_dict['state_name'], self.viewer_id,
-            expected_thread_dict['subject'], 'not used here')
+            self.EXP_ID_1, self.EXPECTED_THREAD_DICT_VIEWER['state_name'],
+            self.viewer_id, self.EXPECTED_THREAD_DICT_VIEWER['subject'],
+            'not used here')
 
-        threads = feedback_services.get_all_threads(self.EXP_ID, False)
+        threads = feedback_services.get_all_threads(self.EXP_ID_1, False)
         self.assertEqual(2, len(threads))
-        self.assertDictContainsSubset(expected_thread_dict,
+        self.assertDictContainsSubset(self.EXPECTED_THREAD_DICT_VIEWER,
                                       threads[1].to_dict())
 
+    def test_get_total_open_threads_before_job_run(self):
+        self.assertEqual(feedback_services.get_total_open_threads(
+            feedback_services.get_thread_analytics_multi([self.EXP_ID_1])), 0)
+
+        feedback_services.create_thread(
+            self.EXP_ID_1, self.EXPECTED_THREAD_DICT['state_name'], None,
+            self.EXPECTED_THREAD_DICT['subject'], 'not used here')
+
+        threads = feedback_services.get_all_threads(self.EXP_ID_1, False)
+        self.assertEqual(1, len(threads))
+
+        self.assertEqual(feedback_services.get_total_open_threads(
+            feedback_services.get_thread_analytics_multi([self.EXP_ID_1])), 0)
+
+    def test_get_total_open_threads_for_single_exploration(self):
```

Comment:
should check that closed threads are not counted/threads that are open and then closed later decrease the count of open threads.